### PR TITLE
Test certificate error card update (DEV)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
Update error card:
1) delete test button added with confirmation dialog
2) update button triggers error dialog after update was unsuccessful

Screenshot 2021-06-09 at 14 18 25

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt [hunk 1]
```diff
@@ -30,12 +31,31 @@ class CovidTestCertificateErrorCard(parent: ViewGroup) :
...
-        retryButton.setOnClickListener { item.onClickAction(item) }
+        retryButton.setOnClickListener {
+            item.onRetryAction(item)
+        }
+
+        if (item.isUpdatingData) {
+            refreshStatus.visibility = View.VISIBLE
+            progressBar.show()
+            retryButton.visibility = View.INVISIBLE
+            deleteButton.visibility = View.INVISIBLE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt:42]: `isInVisible` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt [hunk 2]
```diff
@@ -30,12 +31,31 @@ class CovidTestCertificateErrorCard(parent: ViewGroup) :
             item.testDate.toShortTimeFormat(),
         )
 
-        retryButton.setOnClickListener { item.onClickAction(item) }
+        retryButton.setOnClickListener {
+            item.onRetryAction(item)
+        }
+
+        if (item.isUpdatingData) {
+            refreshStatus.visibility = View.VISIBLE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt:39]: `isVisible`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt [hunk 3]
```diff
@@ -30,12 +31,31 @@ class CovidTestCertificateErrorCard(parent: ViewGroup) :
...
+            progressBar.show()
+            retryButton.visibility = View.INVISIBLE
+            deleteButton.visibility = View.INVISIBLE
+            body.text = context.getString(R.string.test_certificate_error_label_refreshing)
+        } else {
+            refreshStatus.visibility = View.GONE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt:45]: `isGone`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt [hunk 4]
```diff
@@ -30,12 +31,31 @@ class CovidTestCertificateErrorCard(parent: ViewGroup) :
...
+            progressBar.hide()
+            retryButton.visibility = View.VISIBLE
+            deleteButton.visibility = View.VISIBLE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/test/ui/cards/CovidTestCertificateErrorCard.kt:48]: `isVisible`

**Review by mtwalli — COMMENTED** [review]
Works nicely 🚀

**Review by mtwalli — APPROVED** [review]

**Review by d4rken — APPROVED** [review]

**Review by NataliaLemmerth — APPROVED** [review]
UA approved

**Pull request merged**